rest/data/aliases.go

```diff
@@ -9,14 +9,31 @@ import (
 	"github.com/pkg/errors"
 )
 
-// FindProjectAliases queries the database to find all aliases.
-// If the repoId is given, we default to repo aliases if there are no project aliases.
-// If aliasesToAdd are given, then we fold those aliases in and remove any that are marked as deleted.
-// If includeProjectConfig, a merged list of aliases defined on the project page and the project config YAML will be returned,
-// with aliases set on the project page taking precedence.
-func FindProjectAliases(projectId, repoId string, aliasesToAdd []restModel.APIProjectAlias, includeProjectConfig bool) ([]restModel.APIProjectAlias, error) {
-	var err error
-	var aliases model.ProjectAliases
+// FindMergedProjectAliases returns a merged list of aliases, with the order of precedence being:
+// 1. aliases defined on the project page
+// 2. aliases defined on the repo page
+// 3. aliases defined in the project config YAML
+// The includeProjectConfig flag determines whether to include aliases defined in the project config YAML.
+// If the aliasesToAdd parameter is defined, we fold those aliases in and remove any that are marked as deleted.
+func FindMergedProjectAliases(projectId, repoId string, aliasesToAdd []restModel.APIProjectAlias, includeProjectConfig bool) ([]restModel.APIProjectAlias, error) {
+	projectRef, err := model.FindBranchProjectRef(projectId)
+	if err != nil {
+		return nil, errors.Wrapf(err, "finding project ref for project '%s'", projectId)
+	}
+	var projectConfig *model.ProjectConfig
+	if includeProjectConfig {
+		projectConfig, err = model.FindLastKnownGoodProjectConfig(projectId)
+		if err != nil {
+			return nil, errors.Wrapf(err, "finding project config for project '%s'", projectId)
+		}
+	}
+	aliases, err := model.ConstructMergedAliasesByPrecedence(projectRef, projectConfig, repoId)
+	if err != nil {
+		return nil, errors.Wrapf(err, "finding merged aliases for project '%s'", projectId)
+	}
+	if aliases == nil {
```

Comment:
I don't think we want to return early since we mentioned folding in aliasesToAdd.